### DISCUSSION TITLE
SAVING SOME STORAGE FOR POST-PROCESS

### DISCUSSION THREAD

**Discussion by EmreDumlu-UM** [Anura3D/Anura3D_OpenSource#116] [Ideas]
Dear All,

Is there any method to filter the ANURA3D output files to save some storage? Anura3D gives the outputs per timestep, in my case there are more than 12k timestep files. This might result in hardly executing post-process. If I can take those output files at each 10 timestep, It would help me.

I am looking for any ideas. Thanks in advance.

**Comment by albuquerqueraniel** [comment]
Dear Emre,

Anura3D gives the outputs by loadsteps, which are composed of a set of time steps. You can reduce the number of files (and the computational time, since Anura3D doesn't have to write output files at each time step) by changing the value of the "TIME_PER_LOADSTEP" flag in the CPS file.

If you want 10 times fewer files, simply put a value ten times your current time per load step.

Best,
Raniel

**Comment by EmreDumlu-UM** [comment]
Dear Raniel, Thanks. But, increasing time per load step means increasing timestep, isn’t it? I need to keep the timestep as low as for the CFL stability condition.

**Reply by albuquerqueraniel** [reply]
No. We don't change the time step value by changing the TIME_PER_LOADSTEP flag. The time step isn't changed directely in the CPS file, unless you change the courant number. 

The time step value is defined internally by Anura, we don't define it directly. 

When you increase the loadstep time, you just say that those flags defined in the current CPS file are valid during the period of that loadstep. If you don't change any other flag in the CPS file, and increase the loadstep time, you just decrease the frequency that your computer will write the results.

Best regards!

**Reply by Shachen75** [reply]
Will the increase of time step (Time_Per_Loadstep flag) influence the output steps to be reviewed in ParaView? For example, the landslide process per 0.2 time is more smooth than the landslide process calculated from per 0.4 time.

Thanks.

**Reply by ayerrocolom** [reply]
The outputs to be viewed in ParaView are only written at the end of each load step. So, increasing the time per loadstep reduces the frequency of the outputs. For example, if you are only interested in plotting the state after 60 seconds, the time per loadstep can be 60 seconds, and Anura3D will only write the final results. Instead, if you want to track the evolution of the problem every second, the time per loadstep will be 1 for a total number of 60 loadsteps. In the latest, you will have 60 outputs to be viewed in ParaView.

**Comment by EmreDumlu-UM** [comment]
Dear Dr. Alba, Thank you for your reply. I used to consider time per load step as timestep.  Does Increasing the time per load step affect the time step? Due to the CFL condition, we may like to keep the time step as much as minimum to avoid numerical instability.

**Reply by murphy-john** [reply]
The short answer is no, the **time step** is calculated independently of the **time per load step**. It's something that was confusing to me as well when I started with Anura3D. A **load step** is for when you want to input new information or get output. The **time step** is calculated with the CFL condition automatically by Anura3D. If you needed, the easiest way to adjust the **time step** is by adjusting the CFL number in the CPS file input in GiD or the CPS text file. One could also hard code a **time step** in the code (Caution: that should only be done with great care).   

One small exception to that is depending on the timestep automatically calculated by Anura and the **time per load step**, there can be a small change in the _last_ "time step" of a given "load step". The code will run a small time step that is the remainder of **time per load step** divided by **time step**. It does this because the calculated **time per load step** isn't always evenly divisible by  the **time step**. It's usually doesn't come up but I wanted to let you know the load step size can change _a_ time step is possible.

**Comment by EmreDumlu-UM** [comment]
Dear Raniel, Thank you. The first cps file comes with these,  $$NUMBER_OF_LOADSTEPS : 400, $$TIME_PER_LOADSTEP : 1E-003. If the time_per_loadstep flag indicates the frequency of output files, how Anura3D decides to timestep? I thought that we indicate the timestep by defining time per calculation step.   Best regards,

**Comment by EmreDumlu-UM** [comment]
As a result of the conversation with @albuquerqueraniel and @ayerrocolom,

I changed the Courant number from 0.98 to 0.8 and take 

 $$NUMBER_OF_LOADSTEPS
          20
$$TIME_PER_LOADSTEP
   1.00000000000000

which means 20 Anura3D outputs and 20 seconds of total time. The simulation took less time and less storage. 

I really appreciate your answers.  Thank you so much!

**Reply by murphy-john** [reply]
Looks great! If you want to discuss reducing the outputs further, let me know. "Dynamic outputs" is a goal for our project but we haven't had time to implement.

**Comment by EmreDumlu-UM** [comment]
Dear John,

Thank you for your reply. Is there any chance to set up a Zoom meeting at
the end of this month? We would like to discuss what we have simulated so
far and share some ideas.

I also, cc'ed my supervisor, Dr. Ozeren in this mail.

Best regards,

On Wed, Aug 2, 2023 at 9:39 AM John Murphy ***@***.***> wrote:

> The short answer is no, the *time step* is calculated independently of
> the *time per load step*. It's something that was confusing to me as well
> when I started with Anura3D. A *load step* is for when you want to input
> new information or get output. The *time step* is calculated with the CFL
> condition automatically by Anura3D. If you needed, the easiest way to
> adjust the *time step* is by adjusting the CFL number in the CPS file
> input in GiD or the CPS text file. One could also hard code a *time step*
> in the code (Caution: that should only be done with great care).
>
> One small exception to that is depending on the timestep automatically
> calculated by Anura and the *time per load step*, there can be a small
> change in the *last* "time step" of a given "load step". The code will
> run a small time step that is the remainder of *time per load step*
> divided by *time step*. It does this because the calculated *time per
> load step* isn't always evenly divisible by the *time step*. It's usually
> doesn't come up but I wanted to let you know the load step size can change
> *a* time step is possible.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Anura3D/Anura3D_OpenSource/discussions/116#discussioncomment-6617561>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A52I2ZHJL6BWZSHEMSRT6U3XTJRCVANCNFSM6AAAAAAZ7IVVNA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Emre Dumlu, M.Sc., Ph.D. Student
Graduate Research Assistant
National Center for Computational
Hydroscience and Engineering
242 SOC, P.O. Box 1848
University, MS 38677
(Contact): +1 (662) 380 4754
***@***.***
